op-chain-ops/crossdomain/withdrawal.go

```diff
@@ -0,0 +1,112 @@
+package crossdomain
+
+import (
+	"math/big"
+
+	"github.com/ethereum-optimism/optimism/op-bindings/predeploys"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+)
+
+var (
+	// Standard ABI types
+	Uint256Type, _ = abi.NewType("uint256", "", nil)
+	BytesType, _   = abi.NewType("bytes", "", nil)
+	AddressType, _ = abi.NewType("address", "", nil)
+)
+
+// NewWithdrawal will create a Withdrawal
+func NewWithdrawal(
+	nonce *big.Int,
+	sender, target *common.Address,
+	value, gasLimit *big.Int,
+	data []byte,
+) *Withdrawal {
+	return &Withdrawal{
+		Nonce:    nonce,
+		Sender:   sender,
+		Target:   target,
+		Value:    value,
+		GasLimit: gasLimit,
+		Data:     data,
+	}
+}
+
+// Withdrawal represents a withdrawal transaction on L2
+type Withdrawal struct {
+	Nonce    *big.Int
+	Sender   *common.Address
+	Target   *common.Address
+	Value    *big.Int
+	GasLimit *big.Int
+	Data     []byte
+}
+
+// Encode will serialize the Withdrawal so that it is suitable for hashing.
+func (w *Withdrawal) Encode() ([]byte, error) {
+	args := abi.Arguments{
+		{Name: "nonce", Type: Uint256Type},
+		{Name: "sender", Type: AddressType},
+		{Name: "target", Type: AddressType},
+		{Name: "value", Type: Uint256Type},
+		{Name: "gasLimit", Type: Uint256Type},
+		{Name: "data", Type: BytesType},
+	}
+	enc, err := args.Pack(w.Nonce, w.Sender, w.Target, w.Value, w.GasLimit, w.Data)
+	if err != nil {
+		return nil, err
+	}
+	return enc, nil
+}
+
+// EncodeLegacy will serialze the Withdrawal in the legacy format so that it
```

Comment:
Perhaps this should be `LegacyEncode` to be the same as the other legacy functions (where legacy is the prefix) or the others should be updated such that legacy is the suffix